### PR TITLE
Replace nose with pytest.

### PR DESCRIPTION
**Pull request by greut**
The previous PR didn't run any tests... sorry.

For some reason, it fails with 3.2 and 3.7?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ addons:
         # - libwebpmux2
       - libfftw3-dev
       - libglib2.0-dev
+      - python-gobject
```
Comment (jcupitt) [inline .travis.yml]: I'm not sure this is necessary, is it? pyvips does not use gobject-introspection.

Comment (jcupitt) [reply .travis.yml]: I tried without this line and it seems to work OK.

**Comment by jcupitt** [comment]
Yes, the 3.2 and 3.7 fails are puzzling. I found this:

https://github.com/mjcaley/aiospamc/issues/1

so perhaps it'll resolve itself eventually.

**Comment by greut** [comment]
3.2 and [3.3](https://www.python.org/dev/peps/pep-0398/#x-end-of-life) are not maintained anymore. Is it worth keeping those?

**Comment by jcupitt** [comment]
It's harmless to keep it, I think. A few people might still be using it.

**Comment by greut** [comment]
It fails because some eggs are missing the in cheeseshop.

https://pypi.python.org/pypi/setuptools_scm

**Comment by jcupitt** [comment]
OK, I'm persuaded!

**Comment by jcupitt** [comment]
Thank you for fixing this.

**Pull request merged**